### PR TITLE
Update scaling on width change

### PR DESCRIPTION
**Pull request by alisaono**
Update the map scale to fit the width when the map 'width' is changed.

e.g. changing width from 1000 to 500
screen shot 2017-11-04 at 11 48 11 pm
screen shot 2017-11-04 at 11 47 58 pm

### CODE REVIEW

**Comment by alisaono** [comment]
#11 @sanjaypojo @AlmahaAlmalki

**Comment by alisaono** [comment]
This might actually result in a merge conflict with the other branch #20 
You might not want to merge this when you approve... I can merge it locally and push once you approve this pull request.
@sanjaypojo

**Review by sanjaypojo — APPROVED** [review]
Nice, this looks great @alisaono 👍 💯

**Pull request merged**
